### DISCUSSION TITLE
Tectalic Using Dalle

### DISCUSSION THREAD

**Discussion by Wellsj27** [tectalichq/public-openai-client-php#6] [Q&A] [answered]
I am trying to use Dalle as it shows in the documentation:  $openaiClient = Manager::build(new \GuzzleHttp\Client(), new Authentication('$api-key'));
    $response = $openaiClient->imagesGenerations()->create(
      new CreateRequest([
        'model' => '',
        'prompt' => $aiinput,
        'size' => '256x256',
        'n' => 1

      ])
    )->toModel(); But when I run it, it says it needs a model like the completion gets text-davinci this one says it needs one yet there is no model for it. What am I missing?

**Comment by the-csaba** [comment]
Hi there @Wellsj27 ,

I was unable to reproduce the error with the  `ImagesGenerations::create()` method.

Would you mind sharing the following:

- the error message
- `php -v` output
- `composer check-platform-reqs ` output

Thanks

<details>
 <summary>test script</summary>

```php
<?php

declare(strict_types=1);

use Symfony\Component\HttpClient\Psr18Client;
use Tectalic\OpenAi\Authentication;
use Tectalic\OpenAi\Manager;
use Tectalic\OpenAi\Models\ImagesGenerations\CreateRequest;

require_once __DIR__ . '/vendor/autoload.php';

$openaiClient = Manager::build(new Psr18Client(), new Authentication('token'));

$result = $openaiClient->imagesGenerations()->create(new CreateRequest([
    'prompt' => 'draw a puppy',
    'size' => '256x256',
    'n' => 1
]));

var_dump($result->toModel());
```
</details>

**Comment by Wellsj27** [comment]
php -v
PHP 8.1.13 (cli) (built: Nov 24 2022 15:58:42) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.13, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.13, Copyright (c), by Zend Technologies
composer check-platform-reqs
Checking platform requirements for packages in the vendor dir
ext-json 8.1.13   success 
php      8.1.13   success 

![Screenshot 2022-12-20 at 7 45 27 PM (2)](https://user-images.githubusercontent.com/58715629/208808683-3e91ea03-bc35-4264-b483-299ca0191ea4.png)

**Comment by Wellsj27** [comment, marked as answer]
Thanks for the help, I was able to fix my issue, I am just starting out and didn't realize I needed the use use Tectalic\OpenAi\Models\ImagesGenerations\CreateRequest; and couldn't do both image and text in the same code. Thanks again

**Comment by the-csaba** [comment]
Hi  @Wellsj27 ,

I'm glad you figured it out.

Cheers

**Comment by thejamescollins** [comment]
Hi there @Wellsj27,

Just a brief note to say that we have also now updated [our examples documentation](https://tectalic.com/apis/openai/docs/php#examples) to use fully qualified class names.

This change should help users see that different `CreateRequest` models are required for different handlers and methods.

All the best,

James
